crates/fuzzing/src/oracles/dummy/wencoder_generator.rs

```diff
@@ -0,0 +1,239 @@
+    }
+    pub fn export(&mut self, ty: &ExportType<'_>) {
+        let nth = next_index(&mut self.next, &ty.ty());
+        let section_name = format!("item{}", nth);
+
+        let item_ty = ty.ty();
+        self.item(&item_ty);
+
+        let export = extern_to_export(&item_ty, |_| nth);
+        let exports = &mut self.export_section;
+        exports.export(&section_name, export);
+    }
+    fn item(&mut self, ty: &ExternType) {
+        match ty {
+            ExternType::Memory(mem) => {
+                let memories = &mut self.memory_section;
+                memories.memory(wasm_encoder::MemoryType {
+                    limits: Limits {
+                        min: mem.limits().min(),
+                        max: mem.limits().max(),
+                    },
+                });
+            }
+            ExternType::Table(table) => {
+                let tables = &mut self.table_section;
+                tables.table(wasm_encoder::TableType {
+                    element_type: wasm_encoder::ValType::FuncRef,
+                    limits: Limits {
+                        min: table.limits().min(),
+                        max: table.limits().max(),
+                    },
+                });
+            }
+            ExternType::Global(global) => {
+                let globals = &mut self.global_section;
+                globals.global(
+                    wasm_encoder::GlobalType {
+                        val_type: wasm_encoder::ValType::I32,
+                        mutable: global.mutability() == Mutability::Var,
+                    },
+                    value_to_instruction(&global.content()),
+                );
+            }
+            ExternType::Func(ty) => {
+                let types = &mut self.type_section;
+                types.function(
+                    ty.params().into_iter().map(|it| value_to_value(&it)),
+                    ty.results().into_iter().map(|it| value_to_value(&it)),
+                );
+                self.function_section.function(0);
+
+                let locals = vec![];
+                let mut func = Function::new(locals);
+                for ty in ty.results() {
+                    func.instruction(value_to_instruction(&ty));
+                }
+                let codes = &mut self.code_section;
+                codes.function(&func);
+            }
+            ExternType::Module(ty) => {
+                let types = &mut self.type_section;
+                types.module(
+                    ty.imports()
+                        .into_iter()
+                        .map(|x| (x.module(), x.name(), extern_to_entity(&x.ty()))),
+                    ty.exports()
+                        .into_iter()
+                        .map(|x| (x.name(), extern_to_entity(&x.ty()))),
+                );
+
+                let modules = &mut self.module_section;
+                modules.module(&Module::new());
+                modules.module(&Module::new());
```

Comment:
This should be a module with the given module type, not an empty module. This method should create a new module and add the import and export types to that new module's types/imports/exports/etc sections, not the current module's sections.